clevertap-core/src/main/java/com/clevertap/android/sdk/LocalDataStore.java

```diff
@@ -438,7 +440,7 @@ public void run() {
                 }
                 synchronized (PROFILE_FIELDS_IN_THIS_SESSION) {
                     try {
-                        JSONObject profile = dbAdapter.fetchUserProfileById(accountID);
+                        JSONObject profile = dbAdapter.fetchUserProfileByAccountIdAndDeviceID(accountID, deviceInfo.getDeviceID());
```

Comment:
is the device id guaranteed to be present when the code flow comes here?

Reply:
Yess